tls/s2n_prf.c

```diff
@@ -484,29 +496,26 @@ int s2n_prf_calculate_master_secret(struct s2n_connection *conn, struct s2n_blob
  *#                    session_hash)
  *#                    [0..47];
  */
-S2N_RESULT s2n_tls_prf_extended_master_secret(struct s2n_connection *conn, struct s2n_blob *premaster_secret, struct s2n_blob *session_hash)
+S2N_RESULT s2n_tls_prf_extended_master_secret(struct s2n_connection *conn, struct s2n_blob *premaster_secret, struct s2n_blob *session_hash, struct s2n_blob *sha1_hash)
 {
     struct s2n_blob extended_master_secret = {.size = sizeof(conn->secrets.master_secret), .data = conn->secrets.master_secret};
 
     uint8_t extended_master_secret_label[] = "extended master secret";
     /* Subtract one from the label size to remove the "\0" */
     struct s2n_blob label = {.size = sizeof(extended_master_secret_label) - 1, .data = extended_master_secret_label};
 
-    RESULT_GUARD_POSIX(s2n_prf(conn, premaster_secret, &label, session_hash, NULL, NULL, &extended_master_secret));
+    RESULT_GUARD_POSIX(s2n_prf(conn, premaster_secret, &label, session_hash, sha1_hash, NULL, &extended_master_secret));
```

Comment:
With respect to whether including 1.1 and 1.0 makes this change too messy: the fact that we're filling in an existing argument instead of passing NULL makes me lean towards "no". The EMS route is just getting as complicated as the non-EMS route.

However, since you said OpenSSL doesn't support EMS pre-TLS1.2, how are we verifying that our EMS works pre-TLS1.2? Do any of our integration tests use EMS pre-TLS1.2?